src/sagas/backend.ts

```diff
@@ -277,6 +278,56 @@ function* backendSaga(): SagaIterator {
       handleResponseError(resp);
     }
   });
+
+  yield takeEvery(actionTypes.FETCH_NOTIFICATIONS, function*(action) {
+    const tokens = yield select((state: IState) => ({
+      accessToken: state.session.accessToken,
+      refreshToken: state.session.refreshToken
+    }));
+
+    if (!tokens.accessToken) {
+      return;
+    }
+
+    const notifications = yield getNotifications(tokens);
+
+    yield put(actions.updateNotifications(notifications));
+  });
+
+  yield takeEvery(actionTypes.ACKNOWLEDGE_NOTIFICATION, function*(action) {
+    const tokens = yield select((state: IState) => ({
+      accessToken: state.session.accessToken,
+      refreshToken: state.session.refreshToken
+    }));
+    const ids = (action as actionTypes.IAction).payload as number[];
+    const notifications: Notification[] = yield select(
+      (state: IState) => state.session.notifications
+    );
+    const newNotifications: Notification[] = notifications.filter(
+      notification => !ids.includes(notification.id)
+    );
+    yield put(actions.updateNotifications(newNotifications));
```

Comment:
Comment here to indicate why notifications are updated before the server side is done